core/include/detray/propagator/navigator.hpp

```diff
@@ -704,12 +712,30 @@ class navigator {
             ++navigation.next();
             // Release actor constraints
             propagation._stepping.release_step();
+
+            // Surface
+            auto det = navigation.detector();
+            const auto &is = navigation.current();
+            const auto &surface = det->surface_by_index(is->index);
+
             // Update state accordingly
-            navigation.set_state(
-                navigation.volume() != navigation.current()->link
-                    ? navigation::status::e_on_portal
-                    : navigation::status::e_on_module,
-                navigation.current()->index, navigation::trust_level::e_full);
+            switch (surface.id()) {
```

Comment:
This is precisely what I did not want to do every time in the navigation status update, because this increases branching for the entire navigation, instead of only for those actors that need to distinguish a passive from a sensitive. Also, it hard codes the surface ids into the navigator. 
We can add ```is_on_sensitive()``` for the navigation state, if you find that more readable, but please remove the switch case here:

I would suggest:
```
DETRAY_HOST_DEVICE
inline auto is_on_sensitive() const -> bool {
  return _status == navigation::status::e_on_module && current()->id ==  surface_id::e_sensitive;
}
```
If the navigation is not on module, the second predicate (```current()->id ==  surface_id::e_sensitive```)  will not even be checked anymore.